frontend/src/metabase/query_builder/components/view/DataReferenceButton/DataReferenceButton.tsx

```diff
@@ -1,22 +1,29 @@
-/* eslint-disable react/prop-types */
 import { t } from "ttag";
+import cx from "classnames";
+
 import { Icon } from "metabase/core/components/Icon";
 import Tooltip from "metabase/core/components/Tooltip";
-import NativeQuery from "metabase-lib/queries/NativeQuery";
-import { ButtonRoot } from "./DataReferenceButton.styled";
+
+interface DataReferenceButtonProps {
+  className?: string;
+  isShowingDataReference: boolean;
+  size?: number;
+  toggleDataReference: () => void;
+}
```

Comment:
I think we should make every prop except `className` required. It will prevent accidental removal of passing these props - currently they're always present.

This comment applies to `DataReferenceButtonProps`, `NativeVariablesButtonProps` & `SnippetSidebarButtonProps`.